Formula/perl-build.rb

```diff
@@ -126,6 +127,15 @@ def install
     %w[perl-build plenv-install plenv-uninstall].each do |cmd|
       (bin/cmd).write_env_script(libexec/"bin/#{cmd}", PERL5LIB: ENV["PERL5LIB"])
     end
+
+    # Some scripts' shebang is set to the realpath of Homebrew perl, which is not
+    # found using `detected_perl_shebang`. Also, since Formula["perl"].bin returns
+    # the opt_bin directory, we need to manually construct the Cellar path.
```

Comment:
Shouldn't this be fixed in `detected_perl_shebang`?

Reply:
I think the purpose `detected_perl_shebang` is to replace references to `/usr/bin/env perl`, not cellar references to Perl.  I took this comment from a previous PR but I'm not sure it makes sense for this to be handled by `detected_perl_shebang`.

Reply:
`detected_perl_shebang` should be able to fix any shebang, I think. CC @Bo98 for thoughts.

Reply:
Not rewriting `opt` paths is ok since it'll always end up the same end result anyway (the formula should depend on perl and thus end up using opt_bin).

Cellar paths should not exist in the first place. I can have a look where that's coming from.

Though on a related note, the general rewriting of Cellar to opt does falls under the general "should we mass rewrite all external Cellar paths to opt paths?" question I've suggested a couple times before.

Reply:
> Though on a related note, the general rewriting of Cellar to opt does falls under the general "should we mass rewrite all external Cellar paths to opt paths?" question I've suggested a couple times before.

We probably should. I'd even like a formula method you can invoke inside `install` that will do this for every text file in `buildpath`.